cmake/OmimHelpers.cmake

```diff
@@ -2,8 +2,9 @@
 set(OMIM_WARNING_FLAGS
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall>
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wextra>
-  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-unused-parameter>
-  $<$<CXX_COMPILER_ID:AppleClang>:-Wno-deprecated-declarations>  # boost warnings
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wpedantic>
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-unused-parameter>  # We have a lot of functions with unused parameters
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>  # std::wstring_convert and std::codecvt are deprecated in C++17
```

Comment:
1. Is it needed for gcc?
2. Can these specific warning functions be isolated instead of disabling warnings globally?

Reply:
1. Yes.
2. No, there are tons of such varnings in different files. We should fix them and remove these lines later.

Reply:
Will it be possible to distinguish (with compile output) really ill-formed code from some _strange_ warning like "possibly uninitialized variable"?

Reply:
Yes.

Warning:
```
indexer/editable_map_object.cpp:752:29: error: 'codecvt_utf8<char32_t, 1114111, 0>' is deprecated [-Werror,-Wdeprecated-declarations]
  std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
```

Compile-time error:
```
qt/main.cpp:146:3: error: static_assert failed
  static_assert(false);

qt/main.cpp:146:7: error: cannot initialize a variable of type 'int' with an lvalue of type 'const char[7]'
  int x = "asdasd";
```

Reply:
Ok, I have some doubts about -Werror, but we can try. In general, I'm usually against bloating the code making it less readable to fix some strange warning :)

Reply:
Let's rewrite/fix that editable_map_object.cpp code by replacing the deprecated code.

Reply:
It's not possible to fix it now.
There is no solution for it in std and we need to use 3party libs.
Also this is not the only one file with deprecations. There are some deprecation warnings in qt files that can't be fixed now due to this issue: https://github.com/organicmaps/organicmaps/issues/5435

Reply:
It is indeed possible: #5452

Can similar warnings in QT be fixed instead of silencing them? If no, can the silencing option be applied only to the related QT library, or even better, to a few related Qt source files?

Reply:
Please rebase on top of the master and check if we can localize warnings in the qt modules only.